pkg/server/singleprocess/service_runner.go

```diff
@@ -743,7 +743,7 @@ func (s *Service) RunnerJobStream(
 			ws := memdb.NewWatchSet()
 			job, err = s.state(ctx).JobById(ctx, job.Id, ws)
 			if err != nil {
-				errCh <- err
+				errCh <- errors.Wrapf(err, "failed getting job by id from state %q", job.Id)
```

Comment:
Since we're reassigning job on 744, is it possible it's nil here and this might cause a panic?

Reply:
Maybe we should grab the Id as its own var for these err messages